sql/core/src/main/scala/org/apache/spark/sql/sources/commands.scala

```diff
@@ -293,9 +293,18 @@ private[sql] abstract class BaseWriterContainer(
   }
 
   private def newOutputCommitter(context: TaskAttemptContext): OutputCommitter = {
-    outputFormatClass.newInstance().getOutputCommitter(context)
+    val committerClass = context.getConfiguration.getClass(
+      "mapred.output.committer.class", null, classOf[OutputCommitter])
+
+    Option(committerClass).map { clazz =>
+      val ctor = clazz.getDeclaredConstructor(classOf[Path], classOf[TaskAttemptContext])
```

Comment:
Actually, if `committerClass` is based on `mapred` interface, `setupJob` will not work because mapred's output committer use mapred `JobContext` (a subclass of mapreduce's `JobContext`) and we are using `Job` in mapreduce package (another subclass of mapreduce's `JobContext`).